src/modules/auth/auth.guard.ts

```diff
@@ -3,16 +3,23 @@ import { GqlExecutionContext } from '@nestjs/graphql';
 import { AuthGuard } from '@nestjs/passport';
 import { ExecutionContextHost } from '@nestjs/core/helpers/execution-context-host';
 import { AuthenticationError } from 'apollo-server-express';
+import { Reflector } from '@nestjs/core';
 
 @Injectable()
 export class GqlAuthGuard extends AuthGuard('jwt') {
+  constructor(private reflector: Reflector) {
+    super();
+  }
 
   protected readonly logger = new Logger(GqlAuthGuard.name);
 
   canActivate(context: ExecutionContext) {
     const ctx = GqlExecutionContext.create(context);
     const { req } = ctx.getContext();
 
+    const isPublic = this.reflector.get<boolean>('isPublic', context.getHandler());
```

Comment:
Please refer here to my above comment on the public endpoint implementation. In my opinion for public endpoint this Guard should not be applied, instead of by-passing it here using the @isPublic annotation